darts/timeseries.py

```diff
@@ -649,91 +651,113 @@ def from_dataframe(
         TimeSeries
             A univariate or multivariate deterministic TimeSeries constructed from the inputs.
         """
+        df = nw.from_native(df)
 
         # get values
         if value_cols is None:
-            series_df = df.loc[:, df.columns != time_col]
+            series_df = df.drop(time_col) if time_col else df
         else:
             if isinstance(value_cols, str):
                 value_cols = [value_cols]
-            series_df = df[value_cols]
+            series_df = df[value_cols]  # quite slow
```

Comment:
A bit of an edge case, but pandas columns can also be integers, the safest way to deal with that would be:

```py
if isinstance(value_cols, (str, int)):
    value_cols = [value_cols]
series_df = df.select(nw.col(*value_cols))
```